cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -1197,6 +1199,19 @@ TEST_F(TestS3FS, OpenOutputStreamSyncWrites) {
   TestOpenOutputStream();
 }
 
+TEST_F(TestS3FS, OpenOutputStreamNoBucketSanitizationSyncWrites) {
```

Comment:
Instead of adding more test methods for every combinatorial expansion, perhaps we should instead use a `for` loop on the various tested parameter values?

Reply:
I added a struct which abstracts the parameter combinations. [7748824](https://github.com/apache/arrow/pull/41564/commits/77488243b71be6688e9131efac03e9b98d14b1bd)